src/components/Web3Status.tsx

```diff
@@ -15,11 +18,26 @@ const WALLET_VIEWS = {
 }
 
 const Web3Status = (): ReactElement => {
-  const { account } = useWeb3React()
+  const { account, connector } = useWeb3React()
   const [modalOpen, setModalOpen] = useState(false)
   const [walletView, setWalletView] = useState(WALLET_VIEWS.ACCOUNT)
   const { t } = useTranslation()
   const { ensName } = useENS(account)
+  const [user, setUser] = useState<string>("")
+
+  const connectorName = find(SUPPORTED_WALLETS, ["connector", connector])?.name
+  useEffect(() => {
```

Comment:
since all this code is duplicated, let's move it into a hook for easier / cleaner reuse